addons/survey/static/src/js/survey_session_chart.js

```diff
@@ -0,0 +1,245 @@
+odoo.define('survey.session_chart', function (require) {
+'use strict';
+
+var publicWidget = require('web.public.widget');
+var SESSION_CHART_COLORS = require('survey.session_colors');
+
+publicWidget.registry.SurveySessionChart = publicWidget.Widget.extend({
+    init: function (parent, options) {
+        this._super.apply(this, arguments);
+
+        this.questionType = options.questionType;
+        this.questionAnswers = options.questionAnswers;
+        this.questionHasAnswers = options.questionHasAnswers;
+        this.attendeesCount = options.attendeesCount;
+        this.questionStatistics = this._processQuestionStatistics(options.questionStatistics);
+        this.showInputs = options.showInputs;
+        this.showAnswers = false;
+    },
+
+    start: function () {
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self._setupChart();
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Updates the chart data using the latest received question user inputs.
+     *
+     * By updating the numbers in the dataset, we take advantage of the Chartjs API
+     * that will automatically add animations to show the new number.
+     *
+     * @param {Object} updatedQuestionStatistics object containing chart data (counts / labels / ...)
+     */
+    update: function (updatedQuestionStatistics) {
+        if (updatedQuestionStatistics) {
+            this.questionStatistics = this._processQuestionStatistics(updatedQuestionStatistics);
+        }
+
+        if (this.chart) {
```

Comment:
It's a bit complicated.
In summary, yes it could happen to call updateChart when the chart is not initialized yet (but very edge case).


The long story:
In 'survey_session_manage.js', we don't wait for the `_setupChart` to be done before starting to refresh the results every 2 seconds.
So in real edge cases, we could receive the results before the chart is initialized...
Better safe than sorry.